diesel/src/row.rs

```diff
@@ -19,6 +19,12 @@ pub trait Row<DB: Backend> {
     /// would all return `None`.
     fn next_is_null(&self, count: usize) -> bool;
 
+    /// Returns the index of the current column (next to be returned by take, zero based)
+    ///
+    /// May be used to skip the right number of fields when recovering for an `UnexpectedNullError` when
+    /// deserializing an `Option`
+    fn column_index(&self) -> usize;
```

Comment:
As this is a public API I would prefer not to add this method here, because it only has an internal meaning. That said I'm not sure what would be the right approach here.

Reply:
If `Row` has to be public for others to be able to connect new backends I feel like this is close to being the only alternative. Another way I see would be to make sure that when the error is thrown all the fields have been consumed, but that is useless for every other error case and that possibly means a lot of places where we would have to take care of this and possibly make a mistake (enums, structs with Queryable derive,...) .

Given the other functions that this trait requires, this one doesn't seem too absurd to me, especially since the field is already required by most impls.

Reply:
Since it is consistent with `advance`, the meaning is not only internal.

Reply:
I've not forget about this issue, it should definitively be fixed with the diesel 2.0 release, so I've added it the the corresponding mile stone. That said, unfortunately I was pretty occupied with other things in the last time, so I had not really the bandwidth to think much about that. It will be done, but takes probably some time.